### PR TITLE
Python 3: Fix several instances of open function usage

### LINKED ISSUE #9838: threshold_py3_staging cannot install addons
### Steps to reproduce:

* prepare PATH env, as commented in https://github.com/nvaccess/nvda/issues/9831#issuecomment-506375441
* launch installer, create portable copy and run 
* install focusHighlight-6.1.nvda-addon https://github.com/nvdajp/focusHighlight/releases

### Actual behavior:

error occurs as follows

```
INFO - __main__ (14:00:34.233):
Starting NVDA
INFO - external:core.main (14:00:34.623):
Config dir: C:\Applications\nvda_th_py3\userConfig
INFO - external:config.ConfigManager._loadConfig (14:00:34.623):
Loading config: .\userConfig\nvda.ini
INFO - external:core.main (14:00:34.639):
Developer Scratchpad mode enabled
INFO - external:core.main (14:00:34.733):
NVDA version threshold_py3_staging-17851,779ca15b
INFO - external:core.main (14:00:34.733):
Using Windows version 10.0.18362 workstation
INFO - external:core.main (14:00:34.733):
Using Python version 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 21:26:53) [MSC v.1916 32 bit (Intel)]
INFO - external:core.main (14:00:34.733):
Using comtypes version 1.1.7
INFO - external:core.main (14:00:34.733):
Using configobj version 5.1.0 with validate version 1.0.1
INFO - external:synthDriverHandler.setSynth (14:00:36.210):
Loaded synthDriver nvdajp_jtalk
INFO - external:core.main (14:00:36.210):
Using wx version 4.0.3 msw (phoenix) wxWidgets 3.0.5 with six version 1.12.0
INFO - external:brailleInput.initialize (14:00:36.226):
Braille input initialized
INFO - external:braille.initialize (14:00:36.226):
Using liblouis version 3.10.0
INFO - external:braille.initialize (14:00:36.226):
Using pySerial version 3.4
INFO - external:braille.BrailleHandler.setDisplayByName (14:00:36.226):
Loaded braille display driver noBraille, current display has 0 cells.
INFO - external:_UIAHandler.UIAHandler.MTAThreadFunc (14:00:36.468):
UIAutomation: IUIAutomation6
INFO - external:core.main (14:00:36.653):
NVDA initialized
ERROR - external:gui.addonGui.installAddon (14:01:14.624):
Error installing  addon bundle from C:\work\nvda\focusHighlight\focusHighlight-6.1.nvda-addon
Traceback (most recent call last):
  File "C:\projects\nvda\source\gui\addonGui.py", line 552, in installAddon
  File "C:\projects\nvda\source\gui\__init__.py", line 915, in __init__
  File "C:\projects\nvda\source\gui\__init__.py", line 919, in run
  File "C:\projects\nvda\source\addonHandler\__init__.py", line 237, in installAddonBundle
  File "C:\projects\nvda\source\addonHandler\__init__.py", line 294, in __init__
  File "C:\projects\nvda\source\addonHandler\__init__.py", line 695, in __init__
  File "C:\projects\nvda\include\configobj\src\configobj\__init__.py", line 1199, in __init__
  File "C:\projects\nvda\include\configobj\src\configobj\__init__.py", line 1249, in _load
UnicodeDecodeError: 'cp932' codec can't decode byte 0x88 in position 40: illegal multibyte sequence
```

### Expected behavior:

the error should not occur

#### NVDA installed/portable/running from source:

portable

#### NVDA version:

threshold_py3_staging-17851,779ca15b

#### Windows version:

Windows 10 Version 1903, Japanese (ANSI locale i.e. cp932)

#### Have you tried any other versions of NVDA? If so, please report their behaviors.

Python 2 based NVDA should work as expected.

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
Fixes #9838 

### Summary of the issue:
Python can open a file in binary mode and in text mode. In python 2, this difference was less distinct than in Python 3. In python 3, reading a file in text mode returns unicode strings, whereas binary mode returns bytes objects. Therefore, a call of open without supplying proper arguments would open a file in text mode with the default system encoding assumed.

### Description of how this pull request fixes the issue:
This pr fixes the following issues, identified using ``git grep "\Wopen("`

1. #9838: wrong encoding assumed when decoding add-on manifests. As open didn't have an encoding specified, the default text encoding of the system was specified. This failed for Japanese systems.
2. To read Espeak VARIANT DATA, NVDA parses files in the espeak data folder. For this, the open function was also used without an encoding. This resulted in the inability to load espeak on systems where the beta option "Use Unicode UTF-8 for worldwide language support" was enabled.
3. To open calls in NVDAHelper and watchdog are used to open a file without actually writing or reading to it using Python directly. These files are now opened in binary mode. While not strictly necessary, this makes it clearer from a code perspective that we aren't dealing with Python's ability to write or read encoded text files.
4. The logViewer now consistently uses open instead of codecs.open

### Testing performed:
1. Tested that espeak loads
2. Tested that add-ons can install on Japanese systems, see #9838
3. Tested that the log viewer can still show the log and is able to save a copy
4. Tested that nvda_slave still writes its output to the NVDA log of the running instance

### Known issues with pull request:
NOne

### Change log entry:
None

### CODE REVIEW

File: source/synthDrivers/_espeak.py [hunk 1]
```diff
@@ -366,16 +366,19 @@ def getVariantDict():
...
-						name=temp[1].rstrip()
-						break
-				name=None
-			file.close()
+		absFilePath = os.path.join(dir, fileName)
+		if os.path.isfile(absFilePath):
+			# In python 3, open assumes the default system encoding by default.
+			# This fails if Windows' "use Unicode UTF-8 for worldwide language support" option is enabled.
+			# The expected encoding is unknown, therefore use latin-1 to stay as close to Python 2 behavior as possible.
+			with open(absFilePath, 'r', encoding="latin-1") as file:
```
Comment (feerrenrut) [inline source/synthDrivers/_espeak.py]: I don't understand why we wouldn't just open this as UTF-8? With latin-1, we will encounter an error as soon as someone wants to use Japanese/Chinese/Korean characters in a Variant name.

A [stack overflow answer](https://stackoverflow.com/questions/33837551/what-is-the-default-encoding-for-python-2-7-8) points out that `codecs.open` (in Python 2) defaults to ASCII, so if we want the closest behaviour we should use ASCII decoding.

**Review by feerrenrut — COMMENTED** [review]
Overall, this looks good. Thank you for doing this. 

It looks like you have searched for all usages of open?  If so, could you update the description please to specify how you did (include the regex / search method and terms used).

File: source/synthDrivers/_espeak.py (hunk 1 above)
Comment (leonardder) [reply source/synthDrivers/_espeak.py]: The problem is that opening it as utf-8 is the case that fails. It raises an UNicodeDecodeError for an invalid start byte.

The assumption that codecs.open uses ASCII is incorrect, at least on python 2. I tested this by putting an € sign in a file and opening that file with codecs.open on Python 2. It returns python 2 byte strings when reading, so it just returns the raw bytes from the file without applying any decoding.

Note that @michaelDCurran also suggested latin-1 on slack somewhere.

Comment (feerrenrut) [reply source/synthDrivers/_espeak.py]: Ok fair enough. An alternative is to treat the file as binary, process it as bytes, and only decode the b"name" field. Is there any guidance on the format of these files from the espeak project? How would one name a voice with Japanese characters?

Comment (leonardder) [reply source/synthDrivers/_espeak.py]: All errors are due to the usage of \xa0, non breaking space.
I think I prefer leaving this as is, however I will put filing an issue against espeak-ng on my todo list. I think these variant info files really should be utf-8 and non breaking spaces should be removed from them.

**Comment by leonardder** [comment]
@feerenrut: I updated the initial description to tell that I used `git grep "\Wopen("`. Also, I made the espeak case of open log an error on failure with info on what file contained the issue. It looks like there are several of them.

**Review by feerrenrut — APPROVED** [review]
Thanks @leonardder

**Pull request merged**